lending/loan_management/doctype/loan/test_loan.py

```diff
@@ -1465,6 +1468,52 @@ def test_backdated_pre_payment(self):
 		)
 		repayment_entry.submit()
 
+	def test_excess_amount_for_waiver(self):
+		loan = create_loan(
+			"_Test Customer 1",
+			"Term Loan Product 4",
+			100000,
+			"Repay Over Number of Periods",
+			6,
+			"Customer",
+			"2024-07-15",
+			"2024-06-25",
+			10,
+		)
+		loan.submit()
+
+		make_loan_disbursement_entry(
+			loan.name, loan.loan_amount, disbursement_date="2024-06-25", repayment_start_date="2024-07-15"
+		)
+		process_daily_loan_demands(posting_date="2025-01-05", loan=loan.name)
+
+		sales_invoice = frappe.get_doc(
+			{
+				"doctype": "Sales Invoice",
+				"customer": "_Test Customer 1",
+				"company": "_Test Company",
+				"loan": loan.name,
+				"posting_date": "2025-01-15",
+				"posting_time": "00:06:10",
+				"set_posting_time": 1,
+				"items": [{"item_code": "Processing Fee", "qty": 1, "rate": 5000}],
+			}
+		)
+		sales_invoice.submit()
+
+		repayment_entry = create_repayment_entry(loan.name, "2025-01-16", 106684.69)
+		repayment_entry.submit()
+
+		loan_adjustment = frappe.get_doc(
```

Comment:
We are not testing anything in this, please check if charge waiver entry is getting generated if only charge is pending under shortfall acceptance limit.

Also test for Interest and penalty and principal as well